### PR TITLE
Add release notes for v2.1.0

### PR DESCRIPTION
**Pull request by bastbu**
# PR for issue #1589 

## What is being addressed

Release notes for release v2.1.0

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
 title: Release Notes
 ---
 
+## v2.1.0
+
```
Comment (danigian) [inline docs/release-notes.md:6]: Let's add a breaking change section, where we highlight the switch of default deduplication strategy to drop

File: docs/release-notes.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+### New Features
+
+- Enable [multi-gateway scenario supporting higher scalability](./user-guide/scalability.md).
+  This eliminates several disadvantages in a [multi-LNS deployment scenario](./user-guide/deployment-scenarios.md).
+- [Standalone discovery service](./user-guide/lns-discovery.md) for dynamic LNS discovery.
```
Comment (danigian) [inline docs/release-notes.md:11]:
```suggestion
- [Standalone discovery service](./user-guide/lns-discovery.md) for dynamic LNS discovery.
- LoRaBasicsStationModule updated to Basics Station v2.0.6:
  - Support for SX1302 ([configurable](./user-guide/station-module-configuration.md) via "CORECELL" parameter)
  - Adjustable log level ([configurable](./user-guide/station-module-configuration.md) via "LOG_LEVEL" parameter)
```

File: docs/release-notes.md [hunk 3]
```diff
@@ -2,6 +2,28 @@
 title: Release Notes
 ---
 
+## v2.1.0
+
+### New Features
+
+- Enable [multi-gateway scenario supporting higher scalability](./user-guide/scalability.md).
```
Comment (p-schuler) [inline docs/release-notes.md:9]:
```suggestion
- Increase scalability for [multi-gateway scenarios](./user-guide/scalability.md).
```

File: docs/release-notes.md (hunk 1 above)
Comment (bastbu) [reply docs/release-notes.md:6]: Done with 48f3e6f8523ddc751a11262f80185de9cceb7f8f. I will give the new commits another pass once the open PRs are merged.

**Review by MaggieSalak — APPROVED** [review]

File: docs/release-notes.md [hunk 4]
```diff
@@ -2,6 +2,36 @@
...
+## v2.1.0
+
+### New Features
+
+- Increase scalability for [multi-gateway scenarios](./user-guide/scalability.md).
+  This eliminates several disadvantages in a [multi-LNS deployment scenario](./user-guide/deployment-scenarios.md).
+  See [ADR - 010. LNS sticky affinity over multiple sessions](./adr/010_lns_affinity.md).
+- [Standalone discovery service](./user-guide/lns-discovery.md) for dynamic LNS discovery.
+  See [ADR - 009. LoRaWAN Network Server (LNS) discovery](./adr/009_discovery.md).
+- LoRaBasicsStationModule updated to Basics Station v2.0.6:
```
Comment (danigian) [inline docs/release-notes.md:14]:
```suggestion
- LoRaBasicsStationModule v2.1.0 includes:
  - Binaries for Basics Station v2.0.6:
```

Let's try to clarify the confusion brought by mismatch in versioning

@spygi, does this make the situation better?

File: docs/release-notes.md [hunk 5]
```diff
@@ -2,6 +2,37 @@
...
+- [Standalone discovery service](./user-guide/lns-discovery.md) for dynamic LNS discovery.
+  See [ADR - 009. LoRaWAN Network Server (LNS) discovery](./adr/009_discovery.md).
+- LoRaBasicsStationModule v2.1.0 includes:
+  - Binaries for Basics Station v2.0.6:
+  - Support for SX1302 ([configurable](./user-guide/station-module-configuration.md) via "CORECELL" parameter)
+  - Adjustable log level ([configurable](./user-guide/station-module-configuration.md) via "LOG_LEVEL" parameter)
+
+### Breaking Changes
+
+- [#1576](https://github.com/Azure/iotedge-lorawan-starterkit/issues/1576): The default deduplication strategy is now "Drop" instead of "None". More information found in the [decision record](./adr/007_message_deduplication.md).
```
Comment (danigian) [inline docs/release-notes.md:21]: Let's add the new API Version 2022_03_04, which is only compatible with v2.1.0 LNS (Direct method for closing open connections)

Comment (bastbu) [reply docs/release-notes.md:22]: Good idea, done in c72011a7753cc55d596bbcae2ff0cbd39a3acca7

**Pull request merged**